src/lib/sd2/util.py

```diff
@@ -66,7 +66,7 @@ class O(object):
     def our_host_name_inner(o):
         def fn():
             if o.our_host_name is None:
-                o.our_host_name = subprocess.check_output('hostname').rstrip().split('.')[0]
+                o.our_host_name = subprocess.check_output('hostname').decode('ascii').rstrip().split('.')[0]
```

Comment:
What was happening here?

Reply:
The check_output() output is binary and rstrip() fails because it expects a string

Reply:
can you decode('utf-8'), why ascii?

Reply:
hostnames are typically ascii-encoded if I'm not mistaken

Reply:
hostnames are typically ascii-encoded if I'm not mistaken